philander/l6924.py

```diff
@@ -0,0 +1,247 @@
+                St1_params[key.replace("L6924.St1.", '')] = value
+            elif key.startswith("L6924.St2"):
+                St2_params[key.replace("L6924.St2.", '')] = value
+        # check if mandatory parameter are given
+        for param in ["gpio.pinDesignator", "gpio.direction"]:
+            for pin_params in [St1_params, St2_params]:
+                if param not in pin_params.keys():
+                    err = ErrorCode.errInvalidParameter
+                    break
+        # open GPIO pins
+        if err == ErrorCode.errOk:
+            err = self._pinSt1.open(St1_params)
+        if err == ErrorCode.errOk:
+            err = self._pinSt2.open(St2_params)
+        return err
+        
+    def close(self):
+        """Shut down the device after usage.
+        
+        This method should be called when the device is not used, anymore,
+        e.g. as part of the application exit procedure.
+        The following steps are executed:
+        
+        * shut down the device by switching to :attr:`.RunLevel.shutdown`
+        * close serial communication, detach from bus.
+        * close GPIO pins for int1 and int2
+        
+        After return, the device can still be re-used, by calling
+        :meth:`.open` again.
+        
+        Also see: :meth:`.GPIO.close`, :meth:`.Module.close`.
+        """
+        err = self._pinSt1.close()
+        if err == ErrorCode.errOk:
+            err = self._pinSt2.close()
+        return err
+
+    def reset(self):
+        """Soft resets the device.
+        
+        The device is in some default state, afterwards and must be
+        re-configured according to the application's needs.
+        :return: An error code indicating either success or the reason of failure.
+        :rtype: ErrorCode
+        """
+        return ErrorCode.errNotImplemented
+    
+    def getInfo(self):
+        """Retrieves an information block from the charging device.
+        
+        This device does not report any static information.
+
+        :return: The information object and an error code indicating either success or the reason of failure.
+        :rtype: Info, ErrorCode
+        """
+        return ErrorCode.errNotImplemented
+    
+    def isBatteryPresent(self):
+        """Checks, if the battery is present.
+        
+        This does not tell anything about whether the battery is charged
+        or not.
+        
+        Returns :attr:`ErrorCode.errOk` if a battery is present,
+        :attr:`ErrorCode.errUnavailable` if the battery could possibly not be present.\
+        This can not be exactly determined, because there is only one error state,\
+        which could indicate that the battery is not present, but which also could indicate\
+        another problem.
+        
+        See also: :meth:`getChgStatus`.
+
+        :return: An error code.
+        :rtype: ErrorCode
+        """
+        return ErrorCode.errOk if getChgStatus() != Status.unknown else Error.errUnavailable
```

Comment:
Keep the return statement atomic and at the end of the method.
Have a look at the original implementation for a guideline on what to return when.